cpp/src/io/parquet/page_data.cu

```diff
@@ -1025,7 +1028,6 @@ static __device__ bool setupLocalPageInfo(page_state_s* const s,
   if (!t) {
     s->nesting_info = can_use_decode_cache ? s->nesting_decode_cache : s->page.nesting_decode;
   }
-  __syncthreads();
```

Comment:
This `__syncthreads()` still needs to be here, `s->nesting_info` is used by other threads right below this

Reply:
Oops, missed reverting this.